var/spack/repos/builtin/packages/py-torch/package.py

```diff
@@ -175,14 +180,15 @@ class PyTorch(PythonPackage, CudaPackage, ROCmPackage):
     depends_on("py-pybind11@2.10.0", when="@1.13:1", type=("build", "link", "run"))
     depends_on("py-pybind11@2.6.2", when="@1.8:1.12", type=("build", "link", "run"))
     depends_on("py-pybind11@2.3.0", when="@:1.7", type=("build", "link", "run"))
-    depends_on("py-protobuf@3.12.2:", when="@1.10:", type=("build", "run"))
-    depends_on("py-protobuf@:3.14", when="@:1.9", type=("build", "run"))
-    depends_on("protobuf@3.12.2:", when="@1.10:")
-    depends_on("protobuf@:3.14", when="@:1.9")
-    # https://github.com/protocolbuffers/protobuf/issues/10051
-    # https://github.com/pytorch/pytorch/issues/78362
-    depends_on("py-protobuf@:3", type=("build", "run"))
-    depends_on("protobuf@:3", type=("build", "run"))
+    with when("~internal-protobuf"):
+        depends_on("py-protobuf@3.12.2:", when="@1.10:", type=("build", "run"))
+        depends_on("py-protobuf@:3.14", when="@:1.9", type=("build", "run"))
+        depends_on("protobuf@3.12.2:", when="@1.10:")
+        depends_on("protobuf@:3.14", when="@:1.9")
+        # https://github.com/protocolbuffers/protobuf/issues/10051
+        # https://github.com/pytorch/pytorch/issues/78362
+        depends_on("py-protobuf@:3", type=("build", "run"))
+        depends_on("protobuf@:3", type=("build", "run"))
```

Comment:
```suggestion
        depends_on("protobuf@:3")
```
I know you didn't introduce this bug but I definitely think we should be linking to protobuf